core/docz/package.json

```diff
@@ -28,7 +28,7 @@
     "array-sort": "^1.0.0",
     "capitalize": "^2.0.0",
     "docz-components": "2.0.0-rc.48",
-    "docz-core": "2.0.0-rc.51",
+    "docz-core": "file:../docz-core",
```

Comment:
this will not stay. It is a means of testing

Reply:
Could you change this to it only references the latest RC of `docz-core` and push. We could see if the CI tests pass or fail on it.

Reply:
This passes now in circle ci.

My only worry now, since yesterday, is in master the exec command has been moved to the bin node module path of gatsby. This will likely have an effect with this branch trying to find where the bin file is:
https://github.com/doczjs/docz/blob/1e3454fd77caff8678c812bf4fa80cbdc4f62e44/core/docz-core/src/bundler/machine/services/exec-dev-command.ts#L37

Reply:
I will try to find time later today to look at this after I rebase it on top.

Reply:
@rakannimer has been working on that part. Maybe he knows what could go wrong.

Reply:
> My only worry now, since yesterday, is in master the exec command has been moved to the bin node module path of gatsby. This will likely have an effect with this branch trying to find where the bin file is:

I'm sorry I'm not sure I understand, why would your branch cause issues with the change you highlighted ?

Reply:
Basically when I rebase I get this exception as the dependencies are not pulled through via yarn, but instead via npm. Is this a way the modules are resolved into the root directory?

![image](https://user-images.githubusercontent.com/20781479/66332090-d224bf80-e92b-11e9-9d73-419c35db3a6d.png)

(Just a note, when testing, I reference a local version of docz in my package.json which is actually in the core folder like so `file:path-to/docz/core/docz`) Am I making a mistake here? I didnt think so as it seems to work when installing via yarn

Reply:
I think the problem is like this, sorry for not replying better and earlier. We do need to support only having `npm` installed when consuming the packages. That means that we should let user use `npm` when running `docz`. The best solution would be to detect if yarn is installed/used and the use yarn, if not use `npm`. This means that we should let yarn be optional when running as a user: `docz build` and `docz dev`

But internally we still want to use `yarn` to be able to utilize workspaces when building the packages and using lerna. Your PR as it is now changes all occurances of `yarn` to `npm` which also changes how the packages are being resolved and built internally.

Reply:
One way I suppose we could check is the `$npm_execpath`? How do you think we can handle the installing of the packages via npm since workspaces won't work without yarn?